spring-xd-dirt/src/main/java/org/springframework/xd/dirt/server/admin/deployment/zk/ZKDeploymentMessagePublisher.java

```diff
@@ -33,21 +83,164 @@
 	 * @param deploymentQueue the deployment queue
 	 */
 	public ZKDeploymentMessagePublisher(DeploymentQueue deploymentQueue) {
-		this.distributedQueue = deploymentQueue.getDistributedQueue();
+		this.deploymentQueue = deploymentQueue;
+	}
+
+	private CuratorFramework getClient() {
+		return this.deploymentQueue.getClient();
 	}
 
 	/**
-	 * Produces the deployment message into ZK distributed queue.
+	 * Return the amount of time in milliseconds to wait for the message
+	 * consumer to process the deployment message. This value is 3 times
+	 * the value of the configured module deployment timeout.
 	 *
-	 * @param deploymentMessage the deployment message
+	 * @see #deploymentTimeout
+	 * @return timeout for deployment message processing
 	 */
+	private long getTimeout() {
+		return deploymentTimeout * 3;
```

Comment:
I understand why you have chosen a longer timeout than the `deploymentTimeout`. In case when we deploy hundreds or thousands of modules (as many number of module counts), then we might timeout too soon. This also raises one other question if we really want the client to wait for that longer time. But it might makes sense for automated deployments. We may want to have a better strategy to decide the timeout here I believe.